bundles/org.eclipse.e4.emf.xpath/src/org/eclipse/e4/emf/xpath/XPathContextFactory.java

```diff
@@ -38,13 +39,20 @@ public abstract class XPathContextFactory<T> {
 	 * @param parentContext parent context
 	 * @param contextBean   Object
 	 * @return XPathContext
+	 * @deprecated The parent-context does not provide any real value. Just use
+	 *             {@link #newContext(Object)}
 	 */
+	@Deprecated(forRemoval = true, since = "2025-03 (removal in 2027-03 or later)")
 	public abstract XPathContext newContext(XPathContext parentContext, T contextBean);
 
 	/**
 	 * @param <T> the object type the xpath is created for
 	 * @return Create a new XPath-Factory
+	 * @deprecated The implementation returned by this method just supports
+	 *             {@link EObject} elements. Use a specializd factory instead like
+	 *             {@link EcoreXPathContextFactory}.
 	 */
+	@Deprecated(forRemoval = true, since = "2025-03 (removal in 2027-03 or later)")
```

Comment:
For me it looks more that the javadoc is wrong, can we not simply fixup the javadoc:

- remove wrong `@param`
- mention it creates a default instance suitable for query the E4 model (that these are effectively `EObjects` is an implementation detail)

that would best match the intend of this method.

Reply:
That's also an option and it allows to deprecate the specialized `EcoreXPathContextFactory` entirely and to make the generic-type of the `JavaXPathContextFactoryImpl` more correct.